tests/run_tests.py

```diff
@@ -17,8 +17,25 @@ def _test_cpp():
     curr_dir = os.path.dirname(os.path.abspath(__file__))
     build_dir = os.path.join(curr_dir, '../build')
     if os.path.exists(os.path.join(build_dir, cpp_test_filename)):
+        env_used_by_taichi_core = [
+            'HOME',
```

Comment:
Explicitly listing all used envs out seems pretty hard to maintain. As our linux CI is in docker (and windows is moving towards it), maybe it's safe to just copy all envs over to the subprocess?